File: app/src/main/java/com/commit451/teleprinter/sample/MainActivity.java
package com.commit451.teleprinter.sample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

import com.commit451.teleprinter.Teleprinter;

import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

    @OnClick(R.id.button_show)
    public void onShowClicked() {
        mTeleprinter.showKeyboard(mEditText);
    }

    @OnClick(R.id.button_hide)
    public void onHideClicked() {
        mTeleprinter.hideKeyboard();
    }

    @Bind(R.id.edit_text)
    EditText mEditText;

    private Teleprinter mTeleprinter;

    private final Teleprinter.OnKeyboardToggleListener mKeyboardToggleListener = new Teleprinter.OnKeyboardToggleListener() {
        @Override
        public void onKeyboardShown(int keyboardSize) {
            Toast.makeText(MainActivity.this, "keyboard shown with size " + keyboardSize, Toast.LENGTH_SHORT)
                    .show();
        }

        @Override
        public void onKeyboardClosed() {
            Toast.makeText(MainActivity.this, "Keyboard closed", Toast.LENGTH_SHORT).show();
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        mTeleprinter = new Teleprinter(this);
        mTeleprinter.addKeyboardToggleListener(mKeyboardToggleListener);
    }

    @Override
    protected void onDestroy() {
        mTeleprinter.removeKeyboardWatcher(mKeyboardToggleListener);
        super.onDestroy();
    }
}

File: app/src/main/java/com/commit451/teleprinter/sample/MainActivity.kt
package com.commit451.teleprinter.sample

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.commit451.teleprinter.Teleprinter

class MainActivity : AppCompatActivity() {

    lateinit var root: ViewGroup
    lateinit var toolbar: Toolbar
    lateinit var editText: EditText
    lateinit var text: TextView

    lateinit var teleprinter: Teleprinter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        teleprinter = Teleprinter(this, true)

        teleprinter.addOnKeyboardOpenedListener {
            Toast.makeText(this, "Keyboard opened", Toast.LENGTH_SHORT)
                    .show()
            text.text = "Open"
        }
        teleprinter.addOnKeyboardClosedListener {
            Toast.makeText(this, "Keyboard closed", Toast.LENGTH_SHORT)
                    .show()
            text.text = "Closed"
        }
        root = findViewById(R.id.root)
        toolbar = findViewById(R.id.toolbar)
        text = findViewById(R.id.text)
        editText = findViewById(R.id.edit_text)
        editText.append("Hello world")

        toolbar.title = "Teleprinter"
        toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
        toolbar.setNavigationOnClickListener { onBackPressed() }
        findViewById<View>(R.id.button_show).setOnClickListener { teleprinter.showKeyboard(editText) }
        findViewById<View>(R.id.button_hide).setOnClickListener { teleprinter.hideKeyboard() }
        findViewById<View>(R.id.button_is_keyboard_showing).setOnClickListener {
            Toast.makeText(this, if (teleprinter.isKeyboardShowing()) "Yes" else "No", Toast.LENGTH_SHORT)
                    .show()
        }

        teleprinter.showKeyboardWithDelay(editText)
    }

    override fun onDestroy() {
        teleprinter.hideKeyboard()
        super.onDestroy()
    }
}
